packages/babel-parser/src/plugins/jsx/index.js

```diff
@@ -344,9 +344,12 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       if (this.match(tt.braceR)) {
         node.expression = this.jsxParseEmptyExpression();
       } else {
-        node.expression = this.parseExpression();
+        node.expression = process.env.BABEL_8_BREAKING
+          ? this.parseMaybeAssign()
+          : this.parseExpression();
```

Comment:
Consider throw recoverable error if `node.expression` is a `SequenceExpression`.

Reply:
Could we do that as a follow up? :pray: 
I'd like to keep the PR as similar as possible to the one I'm backporting from `next-8-dev`

Reply:
Nah, I can do it here